ports/esp32/Makefile

```diff
@@ -398,17 +403,23 @@ $(SDKCONFIG_COMBINED): $(SDKCONFIG)
 	$(Q)$(MKDIR) -p $(dir $@)
 	$(Q)$(CAT) $^ > $@
 
-$(SDKCONFIG_H): $(SDKCONFIG_COMBINED)
+$(ESPCOMP_KCONFIGS_SOURCE_FILE): $(ESPCOMP_KCONFIGS)
+	$(Q)$(CAT) $^ > $@
+
+$(ESPCOMP_KCONFIGS_PROJBUILD_SOURCE_FILE): $(ESPCOMP_KCONFIGS_PROJBUILD)
+	$(Q)$(CAT) $^ > $@
+
+$(SDKCONFIG_H): $(SDKCONFIG_COMBINED) $(ESPCOMP_KCONFIGS_SOURCE_FILE) $(ESPCOMP_KCONFIGS_PROJBUILD_SOURCE_FILE)
 	$(ECHO) "GEN $@"
 	$(Q)$(MKDIR) -p $(dir $@)
-	$(Q)$(PYTHON) $(ESPIDF)/tools/kconfig_new/confgen.py \
+	$(Q)$(PYTHON2) $(ESPIDF)/tools/kconfig_new/confgen.py \
```

Comment:
Why this change, seems something more related to your build env as opposed to esp-idf v4.0.1 vs. v4.1?